### DISCUSSION TITLE
Extending JSON/YAML language to support JSON References

### DISCUSSION THREAD

**Discussion by whitlockjc** [microsoft/vscode-discussions#118] [Extension Development QnA] [answered]
I would like to create an extension that extends the existing JSON/YAML language support to include features for working with and validating JSON References _(https://github.com/whitlockjc/json-refs)_ in JSON/YAML documents. At a bare minimum, my extension would include a new view for listing JSON References found in the open file/workspace and add support for validating JSON References, but I could see many other features being possible based on the capabilities of the extension API _(hover, references, extract value to reference, reference resolution, ...)_. Based on a [recent discussion](https://github.com/microsoft/vscode-discussions/discussions/101) I feel like I have some questions answered but being completely new to VS Code Extensions, I figured I'd ask the community before jumping in and later realizing I took the wrong path. For example, my first thought is I might need to create a language server but based on the aforementioned discussion, maybe I just need a language client for the existing language servers and use VS Code APIs to drive my extension features? Thank you so much for the documentation and discussions, but I can't find anything that might help me start out on the right foot and avoid going down the wrong path.

**Comment by burkeholland** [comment, marked as answer]
Have you been able to work through this and find the correct path? If so, could you add some context here and mark that as the answer?
